### PR TITLE
Add package plugin

### PR DESCRIPTION
**Pull request by pmuens**
This PR implements the basic building blocks of https://github.com/serverless/serverless/issues/1486

It will hook into the `deploy:createDeploymentPackage` lifecycle and zip the whole service. This .zip file is then stored inside the `.serverless` temporary directory of the service.

The cleanup function which is run in the `after:deploy:deploy` lifecycle will make sure that the `.serverless` directory is removed from the service.

### CODE REVIEW

**Comment by eahefnawy** [comment]
looks like you got everything in the issue covered. Awesome Job 😊

**Comment by pmuens** [comment]
@eahefnawy Thank you! This was kind of a "no brainer" thanks to the great work which was already done and the test coverage we have in place!

File: docs/developing-plugins/building-provider-integrations.md [hunk 1]
```diff
@@ -41,6 +42,11 @@ The purpose of the `deploy:createProviderStacks` lifecycle is to take the basic
 the previous lifecycle and deploy the rough skeleton on the cloud providers infrastructure (without any functions
 or events) for the first time.
 
+#### `deploy:createDeploymentPackage`
+
+The whole service get's zipped up into one .zip file. Serverless will automatically apply exclude rules based on the
+used programming languages (you can always include previously excluded files and folders if you want to).
```
Comment (flomotlik) [inline docs/developing-plugins/building-provider-integrations.md]: At the moment we're not really doing any specific include/exclude rules though are we? Would remove the sentence for now until we know how we're going to do this.

Comment (pmuens) [inline docs/developing-plugins/building-provider-integrations.md]: We do this is the user specifies the `exclude`or `include` rules in the `package` property of the service.

Comment (flomotlik) [inline docs/developing-plugins/building-provider-integrations.md]: But we don't automatically include/exclude based on a specific language. Imho this should be implemented in templates, but not in the packaging mechanism. See this comment for details: https://github.com/serverless/serverless/issues/1517#issuecomment-231380348

Comment (pmuens) [inline docs/developing-plugins/building-provider-integrations.md]: Ah sorry missed that. Will remove it for now. Thanks for the feedback

**Pull request merged**